hw/opentitan/ot_spi_device.c

```diff
@@ -253,59 +255,75 @@ REG32(TPM_WRITE_FIFO, 0x38u)
+ *         | Payload FIFO      | 256B | In  |    | Address FIFO   |  64B |
+ *  0xf00 -+-------------------+------+-----+   -+----------------+------+
+ *         | Command FIFO      |  64B | In  |
+ *  0xf40 -+-------------------+------+-----+
+ *         | Address FIFO      |  64B | In  |
+ *  0xf80 -+-------------------+------+-----+
+ *         | TPM Write buffer  |  64B | In  |
+ *  0xfc0 -+-------------------+------+-----+
  *
  */
 #define SPI_SRAM_READ0_OFFSET 0x0
 #define SPI_SRAM_READ_SIZE    0x400u
+
 #define SPI_SRAM_READ1_OFFSET (SPI_SRAM_READ0_OFFSET + SPI_SRAM_READ_SIZE)
 #define SPI_SRAM_READ1_SIZE   0x400u
-#define SPI_SRAM_MBX_OFFSET   (SPI_SRAM_READ1_OFFSET + SPI_SRAM_READ_SIZE)
-#define SPI_SRAM_MBX_SIZE     0x400u
-#define SPI_SRAM_SFDP_OFFSET  (SPI_SRAM_MBX_OFFSET + SPI_SRAM_MBX_SIZE)
-#define SPI_SRAM_SFDP_SIZE    0x100u
+
+#define SPI_SRAM_MBX_OFFSET (SPI_SRAM_READ1_OFFSET + SPI_SRAM_READ_SIZE)
+#define SPI_SRAM_MBX_SIZE   0x400u
+
+#define SPI_SRAM_SFDP_OFFSET (SPI_SRAM_MBX_OFFSET + SPI_SRAM_MBX_SIZE)
+#define SPI_SRAM_SFDP_SIZE   0x100u
+
+#define SPI_SRAM_TPM_READ_OFFSET (SPI_SRAM_SFDP_OFFSET + SPI_SRAM_SFDP_SIZE)
+#define SPI_SRAM_TPM_READ_SIZE   0x40u
 /* with new scheme (no dual part SRAM, the following offsets are shifted...) */
-#define SPI_SRAM_INGRESS_OFFSET 0x100u
-#define SPI_SRAM_PAYLOAD_OFFSET (SPI_SRAM_SFDP_OFFSET + SPI_SRAM_SFDP_SIZE)
+#define SPI_SRAM_INGRESS_OFFSET 0xE00u
+
+#define SPI_SRAM_PAYLOAD_OFFSET SPI_SRAM_INGRESS_OFFSET
 #define SPI_SRAM_PAYLOAD_SIZE   0x100u
-#define SPI_SRAM_CMD_OFFSET     (SPI_SRAM_PAYLOAD_OFFSET + SPI_SRAM_PAYLOAD_SIZE)
-#define SPI_SRAM_CMD_SIZE       0x40u
-#define SPI_SRAM_ADDR_OFFSET    (SPI_SRAM_CMD_OFFSET + SPI_SRAM_CMD_SIZE)
-#define SPI_SRAM_ADDR_SIZE      0x40u
-#define SPI_SRAM_ADDR_END       (SPI_SRAM_ADDR_OFFSET + SPI_SRAM_ADDR_SIZE)
-#define SPI_SRAM_END_OFFSET     (SPI_SRAM_ADDR_END)
-static_assert(SPI_SRAM_END_OFFSET == 0xe80u, "Invalid SRAM definition");
+
+#define SPI_SRAM_CMD_OFFSET (SPI_SRAM_PAYLOAD_OFFSET + SPI_SRAM_PAYLOAD_SIZE)
+#define SPI_SRAM_CMD_SIZE   0x40u
+
+#define SPI_SRAM_ADDR_OFFSET (SPI_SRAM_CMD_OFFSET + SPI_SRAM_CMD_SIZE)
+#define SPI_SRAM_ADDR_SIZE   0x40u
+
+#define SPI_SRAM_TPM_WRITE_OFFSET (SPI_SRAM_ADDR_OFFSET + SPI_SRAM_ADDR_SIZE)
+#define SPI_SRAM_TPM_WRITE_SIZE   0x40u
+
+#define SPI_SRAM_ADDR_END   (SPI_SRAM_TPM_WRITE_OFFSET + SPI_SRAM_TPM_WRITE_SIZE)
+#define SPI_SRAM_END_OFFSET (SPI_SRAM_ADDR_END)
+static_assert(SPI_SRAM_END_OFFSET == 0xfc0u, "Invalid SRAM definition");
 
 #define SPI_DEVICE_SIZE            0x2000u
 #define SPI_DEVICE_SPI_REGS_OFFSET 0u
 #define SPI_DEVICE_TPM_REGS_OFFSET 0x800u
-#define SPI_DEVICE_SRAM_OFFSET     0x1000u
+#define SPI_DEVICE_SRAM_OFFSET     PARAM_SRAM_OFFSET
 
-#define SRAM_SIZE PARAM_SRAM_OFFSET
+#define SRAM_SIZE 0x1000
 #define EGRESS_BUFFER_SIZE_BYTES \
-    (SPI_SRAM_PAYLOAD_OFFSET - SPI_SRAM_READ0_OFFSET)
+    (SPI_SRAM_INGRESS_OFFSET - SPI_SRAM_READ0_OFFSET)
```

Comment:
Should probably be tightly bounded to better catch writes to unused SRAM?

```suggestion
    (SPI_SRAM_TPM_READ_OFFSET + SPI_SRAM_TPM_READ_SIZE - SPI_SRAM_READ0_OFFSET)
```